platforms/enterprise/enterprise-operations/src/main/java/org/gradle/operations/problems/ProblemUsageProgressDetails.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.operations.problems;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * A problem from the problem API reported via a progress event.
+ *
+ * @since 8.14
+ */
+public interface ProblemUsageProgressDetails {
+
+    /**
+     * The problem definition, i.e. the data that is independent of the usage.
+     *
+     * @since 8.14
+     */
+    ProblemDefinition getDefinition();
+
+    /**
+     * The severity of the problem.
+     * <p>
+     * The severity of a problem is a hint to the user about how important the problem is.
+     * ERROR will fail the build, WARNING will not.
+     *
+     * @since 8.14
+     */
+    String getSeverity();
```

Comment:
💭 Not sure if we should use a String or an enum here.

Reply:
given it's unlikely to change in a non backward compat way
(eg we'd only add potential new severity) I think an enum would be better

Reply:
Enum it is.